tests/certification/secretstores/hashicorp/vault/vault_test.go

```diff
@@ -149,82 +181,397 @@ func TestMultipleKVRetrieval(t *testing.T) {
+		"both",
+		"token mount path and token both set")
+
+	createNegativeTestFlow(fs,
+		"Verify component initialization failure when NEITHER vaultToken nor vaultTokenMountPath are present",
+		"neither",
+		"token mount path and token not set")
+
+	createNegativeTestFlow(fs,
+		"Verify component initialization failure when vaultTokenPath points to a non-existing file",
+		"tokenMountPathPointsToBrokenPath",
+		"couldn't read vault token from mount path")
+
+	createInitSucceedsButComponentFailsFlow(fs,
+		"Verify failure when vaultToken value does not match our servers's value",
+		"badVaultToken",
+		false)
+
+	createPositiveTestFlow(fs,
+		"Verify success when vaultTokenPath points to an existing file matching the configured secret we have for our secret seeder",
+		"tokenMountPathHappyCase",
+		false)
+}
+
+func TestVaultAddr(t *testing.T) {
+	fs := NewFlowSettings(t)
+	fs.secretStoreComponentPathBase = "./components/vaultAddr/"
+	fs.componentNamePrefix = "my-hashicorp-vault-TestVaultAddr-"
+
+	createInitSucceedsButComponentFailsFlow(fs,
+		"Verify initialization success but use failure when vaultAddr does not point to a valid vault server address",
+		"wrongAddress",
+		false)
+
+	createPositiveTestFlow(fs,
+		"Verify success when vaultAddr is missing and skipVerify is true and vault is using a self-signed certificate",
+		"missing",
+		true)
+
+	createPositiveTestFlow(fs,
+		"Verify success when vaultAddr points to a non-standard port",
+		"nonStdPort",
+		true)
+
+	createInitSucceedsButComponentFailsFlow(fs,
+		"Verify initialization success but use failure when vaultAddr is missing and skipVerify is true and vault is using its own self-signed certificate",
+		"missingSkipVerifyFalse",
+		true)
+}
+
+func TestEnginePathCustomSecretsPath(t *testing.T) {
+	const (
+		secretStoreComponentPathBase = "./components/enginePath/"
+		componentNamePrefix          = "my-hashicorp-vault-TestEnginePath-"
+		componentSuffix              = "customSecretsPath"
+		componentName                = componentNamePrefix + componentSuffix
+	)
+
+	currentGrpcPort, currentHttpPort := GetCurrentGRPCAndHTTPPort(t)
+
+	componentPath := filepath.Join(secretStoreComponentPathBase, componentSuffix)
+	dockerComposeClusterYAML := filepath.Join(componentPath, "docker-compose-hashicorp-vault.yml")
+
+	flow.New(t, "Verify success when we set enginePath to a non-std value").
+		Step(dockercompose.Run(dockerComposeProjectName, dockerComposeClusterYAML)).
+		Step("Waiting for component to start...", flow.Sleep(5*time.Second)).
+		Step(sidecar.Run(sidecarName,
+			embedded.WithoutApp(),
+			embedded.WithComponentsPath(componentPath),
+			embedded.WithDaprGRPCPort(currentGrpcPort),
+			embedded.WithDaprHTTPPort(currentHttpPort),
+			// Dapr log-level debug?
+			componentRuntimeOptions(),
+		)).
+		Step("Waiting for component to load...", flow.Sleep(5*time.Second)).
+		Step("✅Verify component is registered", testComponentFound(componentName, currentGrpcPort)).
```

Comment:
Why the checkmark?

Reply:
Leftover from testing - I just wanted to spot some of those error messages quicker.

Let me remove those -- and split this file into 2 or 3 files so tests are in a single place and aux. helper methods are in another one.